### PR TITLE
Remove deprecated RenameActorReminder methods

### LINKED ISSUE #617: Remove deprecated RenameActorReminder methods
The RenameActorReminder methods have been removed in the runtime for Dapr 1.13. The methods need to be removed from the SDK.

See [dapr/dapr#6945](https://github.com/dapr/dapr/pull/6945)

> In most SDKs, methods shouldn't be outright removed, but should return an error saying the method is not implemented instead, to prevent compile-time errors.

### PR DESCRIPTION
**Pull request by dasanind**
Signed-off-by: Anindita Das <dasanuiit@gmail.com>

# Description

Remove deprecated RenameActorReminder methods

## Issue reference
Fixes #617

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
@dasanind 
Thanks for the contribution. Welcome to the community. 
Can you check the issues with the examples and build?

**Comment by dasanind** [comment]
@mukundansundar Can you please approve the build?

**Comment by berndverst** [comment]
@dasanind I'm the maintainer of the repo. Can you please tell me which version of grpctools you used? I'd like the protos to be generated with at least version `1.54.0` or higher. I see a lot of differences in the generates protos here that should not exist, so please check the version.

File: dapr/proto/common/v1/common_pb2.py [hunk 1]
```diff
@@ -26,30 +27,30 @@
...
-  _STATEOPTIONS._serialized_end=1212
-  _STATEOPTIONS_STATECONCURRENCY._serialized_start=1009
-  _STATEOPTIONS_STATECONCURRENCY._serialized_end=1113
-  _STATEOPTIONS_STATECONSISTENCY._serialized_start=1115
-  _STATEOPTIONS_STATECONSISTENCY._serialized_end=1212
-  _CONFIGURATIONITEM._serialized_start=1215
-  _CONFIGURATIONITEM._serialized_end=1388
-  _CONFIGURATIONITEM_METADATAENTRY._serialized_start=772
-  _CONFIGURATIONITEM_METADATAENTRY._serialized_end=819
+  _globals['_HTTPEXTENSION']._serialized_start=87
```
Comment (berndverst) [inline dapr/proto/common/v1/common_pb2.py:30]: This is indicative of a very different version `grpcio-tools`.

**Review by berndverst — CHANGES_REQUESTED** [review]
Please install `grpcio-tools>=1.54.0` then generate the protos again.

File: dapr/proto/common/v1/common_pb2.py (hunk 1 above)
Comment (berndverst) [reply dapr/proto/common/v1/common_pb2.py:30]: I just checked against version 1.60 and that seems to be using `_globals` whereas 1.54 does not. That would be fine for me, but please confirm what version you actually used.

Comment (berndverst) [reply dapr/proto/common/v1/common_pb2.py:30]: When you generated this, did you also get a line like `# Protobuf Python Version: 4.25.0` in the output? Feel free to leave that in.

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
I pushed a commit with the latest generated protos using 1.60.0 tools.

File: dapr/proto/common/v1/common_pb2.py (hunk 1 above)
Comment (dasanind) [reply dapr/proto/common/v1/common_pb2.py:30]: @berndverst Looks like the build passed. Do you need me to look at the above queries?

**Comment by berndverst** [comment]
Technically this PR does not address #617 because that issue itself requires no action. We did not expose the RenameActorReminder in any way.

Everything Actor related in the Python SDK uses HTTP - not protos. See `clients/http/dapr_actor_http_client.py`.

But we have to updates protos ahead of each Dapr release anyway. So I'll merge this PR so it can count as a contribution for you.

**Comment by dasanind** [comment]
Thank you @berndverst

**Pull request merged**